storage/engine/rocksdb/db.cc

```diff
@@ -39,33 +39,84 @@
 extern "C" {
 #include "_cgo_export.h"
 
-struct DBBatch {
+struct DBEngine {
+  rocksdb::DB* const rep;
```

Comment:
I ended up leaving this pointer because it is acting like an accessor for the `DB{Flush,CompactRange,etc}` methods. I renamed `EngineGetter` to `DBGetter` and cleaned up the implementation slightly. `MakeReadOptions` was removed in favor of a `read_opts` member inside `DB{Impl,Batch,Snapshot}`.